feat(CODEOWNERS): Add sustaining team as nats* package owners

philroche opened the pull request:
With the possible move in licensing for the nats* packages to a less open license this commit
add sustaining team as codeowner thus blocking any non sustaining team merges of nats* packages
and also blocks automation from any automated updates or merges of nats* package updates.

This change is likley temporary until we can establish that the move to less open license is not proceeding.

Signed-off-by: philroche <phil.roche@chainguard.dev>

philroche committed:
feat(CODEOWNERS): Add sustaining team as nats* package owners

With the possible move in licensing for the nats* packages to a less open license this commit
add sustaining team as codeowner thus blocking any non sustaining team merges of nats* packages
and also blocks automation from any automated updates or merges of nats* package updates.

This change is likley temporary until we can establish that the move to less open license is not proceeding.

Signed-off-by: philroche <phil.roche@chainguard.dev>

murraybd commented on CODEOWNERS:
```diff
@@ -25,3 +25,4 @@ glibc.yaml           @wolfi-dev/foundations-squad
 llvm*.yaml           @wolfi-dev/foundations-squad
 openssh.yaml         @wolfi-dev/foundations-squad
 openssl.yaml         @wolfi-dev/foundations-squad
+nats*.yaml           @wolfi-dev/sustaining-team
```
Adding a comment here would be helpful for future travellers.

philroche replied:
Done. Thank you

IdlePhysicist reviewed (dismissed):
Brian makes a good point.  Though if you squash merge the commit will refer back to the PR at least

philroche committed:
feat(CODEOWNERS): Add sustaining team as nats* package owners - inline comments added

Add inline comments to CODEOWNERS with reasoning for why nats* packages are now owned by sustaining team

Signed-off-by: philroche <phil.roche@chainguard.dev>
